modules/core/src/main/java/org/apache/fluo/core/impl/TransactionImpl.java

```diff
@@ -596,9 +597,14 @@ public CommitData createCommitData() {
 
   @Override
   public synchronized void commit() throws CommitException {
-    SyncCommitObserver sco = new SyncCommitObserver();
-    commitAsync(sco);
-    sco.waitForCommit();
+    SyncCommitObserver sco = null;
+    try {
+      sco = new SyncCommitObserver();
+      commitAsync(sco);
+      sco.waitForCommit();
+    } finally {
+      sco = null;
```

Comment:
Thanks for the article.  Of those you suggest, I can only null the weakNotifications.  The rest are final, but I can clear them since they are collections.  I'll push a change for review.